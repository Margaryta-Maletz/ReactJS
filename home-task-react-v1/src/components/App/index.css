.wrapper {
    width: 1200px;
    box-sizing: border-box;
    padding: 20px 60px;
    display: flex;
}
