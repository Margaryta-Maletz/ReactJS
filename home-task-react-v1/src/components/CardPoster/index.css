.card-poster_wrapper {
    width: 322px;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    font-family: Montserrat;
    font-style: normal;
    font-weight: 500;
    color: white;
    mix-blend-mode: normal;
}

.card-poster_title {
    font-size: 18px;
    line-height: 22px;
    opacity: 0.7;
    margin: 22px 0 8px;
    width: 75%;
}

.card-poster_genre {
    font-size: 14px;
    line-height: 17px;
    opacity: 0.5;
    margin-block: 0;
    margin: 0 0 50px;
}

.card-poster_release-date {
    width: 66px;
    height: 26px;
    border: 1px solid #555555;
    box-sizing: border-box;
    border-radius: 4px;
    font-size: 14px;
    line-height: 24px;
    text-align: center;
    margin: 22px 0 8px;
}
