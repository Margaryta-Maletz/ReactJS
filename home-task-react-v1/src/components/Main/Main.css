.main {
    position: relative;
    background-color: #232323;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    column-gap: 20px;
}

.top-border.wrapper {
    position: absolute;
    width: 1080px;
    height: 45px;
    border-bottom: 3px solid #555555;
    z-index: -1;
}

