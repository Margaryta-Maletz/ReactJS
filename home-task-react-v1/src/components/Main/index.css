.main {
    position: relative;
    background-color: #232323;
    z-index: -1;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    column-gap: 300px;
}

.top-border.wrapper {
    position: absolute;
    width: 1080px;
    height: 45px;
    border-bottom: 3px solid #555555;
    z-index: -1;
}

