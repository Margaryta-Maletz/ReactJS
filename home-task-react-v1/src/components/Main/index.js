import React from 'react';
import { GenreToggle } from '../GenreToggle';
import {ResultsSort} from '../ResultsSort';
import { ResultCount } from '../ResultCount';
import { CardPosterList } from '../CardPosterList';
import './index.css';
import ErrorBoundary from "../ErrorBoundary";

const Main = () => {
    const genres = ['All', 'Documentary', 'Comedy', 'Horror', 'crime'];
    const sortList = ['Release Date', 'title', 'genre'];

    return (
        <main className='main wrapper'>
            <div className='top-border wrapper'></div>
            <ErrorBoundary>
                <GenreToggle genres = { genres } />
                <ResultsSort sort = { sortList } />
                <ResultCount count='49' />
                <CardPosterList />
            </ErrorBoundary>
        </main>
    )
}

export default Main;
