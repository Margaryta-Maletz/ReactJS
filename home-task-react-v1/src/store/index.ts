import thunk from 'redux-thunk';
import {configureStore, getDefaultMiddleware, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './slice';

const middleware = getDefaultMiddleware({
    imitableCheck: false,
    serializableCheck: false,
    thunk: true,
});

export const store = configureStore({
    reducer: rootReducer,
    middleware: middleware,
    devTools: process.env.NODE_ENV !== 'production',
});

/*export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;*/
