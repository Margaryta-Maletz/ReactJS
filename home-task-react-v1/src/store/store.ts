import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import rootReducer from './reducers';

export const store = createStore(
    rootReducer,
    applyMiddleware(thunk)
);

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> =
    ThunkAction<
        ReturnType,
        RootState,
        unknown,
        Action<string>
    >;
