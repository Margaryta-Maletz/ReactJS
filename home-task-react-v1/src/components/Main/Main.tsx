import React, {useEffect} from 'react';
import { GenreToggle } from '../GenreToggle';
import { ResultsSort } from '../ResultsSort';
import { ResultCount } from '../ResultCount';
import { CardPosterList } from '../CardPosterList';
import styles from './Main.module.css';
/*import {useDispatch, useSelector} from "react-redux";*/
import {IMovie} from "../../store/types";
// @ts-ignore
/*import ErrorBoundary from "../ErrorBoundary";*/
/*import {useHistory, useLocation, useParams} from "react-router-dom";
import {Genre, genres, SortList, sortListItems} from "../../consts";
import {fetchMovies, getMoviesStart, setActiveGenre, setSearchString, setSortItem} from "../../store/slice";*/
interface MainProps {
    totalAmount: number,
    movies: IMovie[],
}
export const Main:React.FC<MainProps> = ({totalAmount, movies}) => {
/*    const { activeGenre, sortItem, editMovieList } = useSelector((state: IState) => state);
    const { totalAmount } = useSelector((state: IState) => state);
    const dispatch = useDispatch();
    const location = useLocation();
    const history = useHistory();
    const { searchQuery = '' } = useParams<{searchQuery?: string}>();*/

    useEffect(() => {
/*        const params = new URLSearchParams(location.search);
        const sortBy = params.get('sortBy');
        const genre = params.get('genre');
        genre && genres.includes(genre as Genre) ? dispatch(setActiveGenre(genre)) : params.delete('genre');
        sortBy && sortListItems.includes(sortBy as SortList) ? dispatch(setSortItem(sortBy)) : params.delete('sortBy');
        dispatch(setSearchString(searchQuery));
        params && history.push(`?${params.toString()}`);*/
    }, []);

/*    useEffect(() => {
        dispatch(getMoviesStart());
        dispatch(fetchMovies());
    }, [activeGenre, sortItem, editMovieList]);*/

    return (
        <main className={`wrapper ${styles.main}`}>
            <div className={`wrapper ${styles.topBorder}`}/>
            <GenreToggle />
            <ResultsSort sortItem={''}/>
            <ResultCount count={ totalAmount.toString() } />
            <CardPosterList movies={movies}/>
        </main>
    )
}
