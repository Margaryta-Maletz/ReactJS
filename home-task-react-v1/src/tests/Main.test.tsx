import React from "react";
import { render } from '@testing-library/react';
import { Main } from '../components/Main/';
import { Provider } from "react-redux";
import { store } from "../store/store";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "localhost:8080/search/"
    }),
    useHistory: (path: string) => ({
        push: jest.fn(),
        location: {
            pathname: "localhost:8080/search/" + path
        }
    }),
    useParams: jest.fn().mockReturnValue({ searchQuery: '' }),
}));

jest.mock('../components/ErrorBoundary/');

describe('make snapshot', () => {
    test('renders', () => {
        const { container } = render(
            <Provider store={store}>
                <Main />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});
