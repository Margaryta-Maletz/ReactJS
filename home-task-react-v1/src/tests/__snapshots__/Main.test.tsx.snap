// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot renders 1`] = `
<div>
  <main
    class="main wrapper"
  >
    <div
      class="top-border wrapper"
    />
  </main>
</div>
`;
